cmake/SetupCodeChecks.cmake

```diff
@@ -45,122 +45,93 @@
+## blt_add_uncrustify_target( NAME              <Created Target Name>
+##                            MODIFY_FILES      [TRUE | FALSE (default)]
+##                            CFG_FILE          <Uncrustify Configuration File> 
+##                            FLAGS             <Additional Flags to Uncrustify>
+##                            COMMENT           <Additional Comment for Target Invocation>
+##                            WORKING_DIRECTORY <Working Directory>
+##                            REDIRECT          <Redirection Command>
+##                            SRC_FILES         [FILE1 | [FILE2 ...]] )
 ##
-## blt_add_code_check_targets(cfg)
+## Creates a new target with the given NAME for running uncrustify over the given SRC_FILES.
 ##
-##------------------------------------------------------------------------------
-macro(blt_add_code_check_targets cfg_file)
-
-    if(UNCRUSTIFY_FOUND)
-        # Only run uncrustify on C and C++ files
-        # Note, we can later extend this by passing in a list of valid types to the macro
-        set(_fileTypes ".cpp" ".hpp" ".cxx" ".hxx" ".cc" ".c" ".h" ".hh")
-    
-        # generate the filtered list of source files
-        set(_filt_sources)
-        foreach(_file ${${PROJECT_NAME}_ALL_SOURCES})
-          get_filename_component(_ext ${_file} EXT)
-          list(FIND _fileTypes "${_ext}" _index)
-      
-          if(_index GREATER -1)
-             file(RELATIVE_PATH _relpath ${CMAKE_CURRENT_BINARY_DIR} ${_file})
-             list(APPEND _filt_sources ${_relpath})
-          endif()
-        endforeach()
-
-        blt_add_uncrustify_check(CFG_FILE ${cfg_file}   SRC_FILES ${_filt_sources})
-        blt_add_uncrustify_inplace(CFG_FILE ${cfg_file} SRC_FILES ${_filt_sources})
-    endif()
-    
-endmacro()
-    
-
-##------------------------------------------------------------------------------
-## - Macro for invoking uncrustify to check code formatting
+## MODIFY_FILES, if set to TRUE, modifies the files in place and adds the created target to
+## the style target.  Otherwise the files are not modified and the created target is added
+## to the check target.
+##
+## CFG_FILE defines the uncrustify settings.
 ##
-## blt_add_uncrustify_check( CFG_FILE <uncrustify_configuration_file> 
-##                           FLAGS <additional_flags_to_uncrustify>
-##                           COMMENT <additional_comment_for_target_invocation>
-##                           WORKING_DIRECTORY <working_directory>
-##                           REDIRECT <redirection_commands>
-##                           SRC_FILES <list_of_src_files_to_uncrustify> )
+## FLAGS are additional flags given to uncrustify.
+##
+## COMMENT is prepended to the commented outputted by CMake.
+##
+## WORKING_DIRECTORY is the directory that uncrustify will be ran.  It defaults to the directory
+## where this macro is called.
+##
+## REDIRECT is appended to the uncrustify command and allows you to redirect output to a file.
```

Comment:
Thanks for adding this.  Should ``REDIRECT`` just be ``EXTRA_COMMANDS``  or ``ADDL_OPTIONS`` or something like that?  You're basically just adding it to the uncrustify invocation.